TombEngine/Game/effects/effects.cpp

```diff
@@ -1475,22 +1282,22 @@ void TriggerFireFlame(int x, int y, int z, int fxObj, int type)
 
 		spark->on = true;
 
-		if (type == 2)
+		if (type == FlameType::Stove)
 		{
 			spark->sR = spark->sG = (GetRandomControl() & 0x1F) + 48;
 			spark->sB = (GetRandomControl() & 0x3F) - 64;
 		}
 		else
 		{
-			if (type == 254)
+			if (type == FlameType::StoveHigh)
 			{
 				spark->sR = 48;
-				spark->sG = 255;
-				spark->sB = (GetRandomControl() & 0x1F) + 48;
+				spark->sG = 48;
+				spark->sB = (GetRandomControl() & 0x1F) + 128;
```

Comment:
How do these `(GetRandomControl() & 0xsomething) + something` calculations work exactly, and is there a chance they can be substituted with a more generic `Random::GenerateInt(minValue, maxValue)` in the future?

Reply:
Yes, absolutely, in fact I am gradually converting those I encounter, I guess it can be done in this case too.